### ISSUE TITLE
Linting symlinked .yaml files?

### ISSUE THREAD

**Issue by Iry-Horus** [stoplightio/spectral#1480]
Hey guys,

Using Spectral to implement linting on a client project, but we have an automation testing framework in the same repo as the core app framework. Some of the OpenAPI definitions (.yaml files) are in the core app but symlinked in the automation testing framework. As such, when I lint against the folder with all the .yaml files in it inside of the automation testing framework Spectral throws `invalid-ref  ENOENT: no such file or directory` on the reference paths for the symlinked files.

Does anyone have an elegant solution to this? I can't just lint the whole core app for .yaml files. Should I try converting the .yaml files to strings then 'lint as a string,' or is there a better/easier way?

Any help would be greatly appreciated, and thanks for making Spectral -- it's awesome!

**Comment by P0lip** [comment]
Hey!
Did you try providing a custom resolver?
Here's how https://github.com/stoplightio/spectral/blob/e6ba6a88932f7161313018fbfdaffa6185ea15cb/docs/guides/3-javascript.md#using-a-custom-resolver
You could tweak the logic to support symlinks.

If you are a CLI user, you can provide the instance via `--resolver` flag

```js
// my-resolver.js
const path = require("path");
const fs = require("fs");

const { Resolver } = require("@stoplight/json-ref-resolver");

module.exports = new Resolver({
  resolvers: {
      file: {
      resolve: ref => {
        return new Promise((resolve, reject) => {
          const basePath = process.cwd();
          const refPath = ref.path();
          fs.readFile(path.join(basePath, refPath), "utf8", (err, data) => {
            if (err) {
              reject(err);
            } else {
              resolve(data);
            }
          });
        });
      },
    },
  },
});
```
Do note, however, that the above will override the existing resolver, meaning http and https protocol might be not supported.
If you need to preserve that capability, let me know and we can try to adjust the usage.

**Comment by P0lip** [comment]
I'll close this issue for now. Do not hesitate to reach out if you still experience issues!
